src/test/java/org/openrewrite/java/migrate/lang/UseTextBlocksTest.java

```diff
@@ -393,4 +392,78 @@ public static void main(String[] args) {
+     *                 + "\n                                                         "
+     *                 + "\n=========================================================";
+     *         String s2 = """
+     *                     =========================================================
+     *                                                                             \s
+     *                               Welcome to Spring Integration!                \s
+     *                                                                             \s
+     *                         For more information please visit:                  \s
+     *                         https://www.springsource.org/spring-integration     \s
+     *                                                                             \s
+     *                     =========================================================\
+     *                     """;
+     *         String s3 = """
+     *                     \n=========================================================\
+     *                     \n                                                         \
+     *                     \n          Welcome to Spring Integration!                 \
+     *                     \n                                                         \
+     *                     \n    For more information please visit:                   \
+     *                     \n    https://www.springsource.org/spring-integration      \
+     *                     \n                                                         \
+     *                     \n=========================================================\
+     *                     """;
+     */
+    @Test
+    void newlineAtBeginningOfLines() {
+        rewriteRun(
+          version(
+            java(
+              """
+                class A {
+                    void welcome() {
+                        log("\\n========================================================="
+                            + "\\n                                                         "
+                            + "\\n          Welcome to Spring Integration!                 "
+                            + "\\n                                                         "
+                            + "\\n    For more information please visit:                   "
+                            + "\\n    https://www.springsource.org/spring-integration      "
+                            + "\\n                                                         "
+                            + "\\n=========================================================");
+                    }
+                    void log(String s) {}
+                }
+                """,
+              """
+                class A {
+                    void welcome() {
+                        log(\"""
+                           \s
```

Comment:
This line looks wrong to me. This will print an `s`. Also note that in the result after the recipe every line has a leading space, whereas in the input there is no leading space.

Reply:
I think this is correct, `\s` here is used to explicitly make the expected result string preserve leading white spaces.

Reply:
the input has not a leading space, but a leading new line, and that new line is converted to indentation (those white spaces)  + newline.